pkg/controller/cloudmetrics/cloudmetrics_controller.go

```diff
@@ -218,6 +281,22 @@ func (r *ReconcileCloudMetrics) Reconcile(request reconcile.Request) (reconcile.
 			scrapedMetrics = append(scrapedMetrics, scrapedMetricsOutput.Metrics...)
 		}
 	}
+	// for each scraped metric value we check redisGaugeMetrics for a match and set the value and labels
+	for _, scrapedMetric := range scrapedMetrics {
+		for _, croMetric := range redisGaugeMetrics {
+			if scrapedMetric.Name == croMetric.Name {
+				croMetric.GaugeVec.WithLabelValues(
+					scrapedMetric.Labels[labelClusterIDKey],
+					scrapedMetric.Labels[labelResourceIDKey],
+					scrapedMetric.Labels[labelNamespaceKey],
+					scrapedMetric.Labels[labelInstanceIDKey],
+					scrapedMetric.Labels[labelProductNameKey],
+					scrapedMetric.Labels[labelStrategyKey]).Set(scrapedMetric.Value)
+				r.logger.Infof("successfully set redis metric value for %s", croMetric.Name)
+				continue
+			}
+		}
+	}
```

Comment:
Would it be worth adding a helper function here, as this logic is repeated for `postgresGaugeMetrics`

Reply:
it might